protocol/message.go

```diff
@@ -208,3 +208,30 @@ func NewMonitoringProof(ap []*m.AuthenticationPath,
 		},
 	}, ReqSuccess
 }
+
+// GetKey returns the key extracted from
+// a _verified_ CONIKS server's response.
+//
+// If the response contains a single authentication path,
+// the key is obtained from that authentication path or the
+// temporary binding (which depends on the returned proof type).
+//
+// If the response contains a range of authentication paths,
+// the key is obtained from the authentication path corresponding
+// with the most recent signed tree root.
+func (msg *Response) GetKey() []byte {
```

Comment:
Should we somehow assert that this is only called on responses that have been through,

```
	if Errors[msg.Error] {
		return msg.Error
	}
	if err := validateResponse(requestType, msg); err != nil {
		return err.(ErrorCode)
	}
```